pkg/cluster/cluster.go

```diff
@@ -346,21 +320,27 @@ func (c *Cluster) Create() error {
 	//
 	// Do not consider connection pooler as a strict requirement, and if
 	// something fails, report warning
-	roles := c.RolesConnectionPooler()
-	for _, r := range roles {
-		if c.ConnectionPooler != nil {
-			c.logger.Warning("Connection pooler already exists in the cluster")
-			return nil
+	if c.needConnectionPooler() {
+
+		roles := c.RolesConnectionPooler()
+		for _, r := range roles {
+			c.logger.Warningf("found roles are %v", r)
 		}
-		connectionPooler, err := c.createConnectionPooler(c.installLookupFunction)
-		if err != nil {
-			c.logger.Warningf("could not create connection pooler: %v", err)
-			return nil
+
+		for _, r := range c.RolesConnectionPooler() {
+			if c.ConnectionPooler[r] != nil {
+				c.logger.Warning("Connection pooler already exists in the cluster")
```

Comment:
It would be more descriptive here to log also the role for which the pooler exists, and the name of this pooler similar to how errors are processed below in this code